compare-evidence-strings/compare.sh

```diff
@@ -1,130 +1,210 @@
 #!/bin/bash
 # A script to compare evidence strings. Please see README.md for details on using it.
 
-# A function to sort key in evidence strings. This makes them more readable and helps comparison through word diff.
-# Also removes the "validated_aginst_schema_version" field entirely, because it frequently changes between the
-# versions, and this change is not important.
+
+
+########################################################################################################################
+echo "Defining functions"
+
+# A function to sort keys in the evidence strings. This makes them more readable and helps comparison through word diff.
+# Also removes several version- and date-related fields which are changed frequently, but do not reflect actual change:
+# * validated_aginst_schema_version
+# * date_asserted
 # The two arguments are input and output JSON files.
 sort_keys () {
   ./jq -S "." --tab <"$1" \
     | tr -d '\t\n' \
     | sed -e 's|}{|}~{|g' \
     | tr '~' '\n' \
     | sed -e 's|,"validated_against_schema_version": "[0-9.]*"||g' \
+    | sed -e 's|"date_asserted": ".\{19\}",||g' \
```

Comment:
Is the set of such non-informative attributes a moving target based on how often the JSON schema evolves (which I guess should be infrequent but I was curious since it has changed since the last time)?

Reply:
No, this is also not a moving target (at least for our use case). There was no change, really; rather, I only ran the previous protocol version on the _same_ input ClinVar data, but different code versions. Once I started running this version (v3) between ClinVar versions, I noticed that this field is also not informative and needs to be excluded.

Reply:
In the future, depending on how much Open Targets will be interested in this, we might make this protocol make more generic and those fields and URLs will be modifiable parameters; however, the current approach is fine for our use case